src/analysis/history.hpp

```diff
@@ -71,6 +76,59 @@ Real ReduceOneVar(MeshData<Real> *md, const std::string &varname, int idx = 0) {
   return result;
 }
 
+template <typename Reducer_t>
```

Comment:
If we're always going to sum, you probably don't need to template on reduction here.